### ISSUE TITLE
Rich Editor - Menus (Escaping and Accessibility)

### ISSUE THREAD

**Issue by slafleche** [vanilla/vanilla#6404]
When highlighting a word or a paragraph, show contextual menu for highlighted text.

Use [Popper.js](https://popper.js.org/) for handling the position of the menu. This will make it easier to keep responsive.

**Comment by slafleche** [comment]
Markup will come from: #6408

**Comment by slafleche** [comment]
The tab order needs to be respected through the menu. 

When you've got text selected, tabbing should bring you to the first element in the menu. From there you should be able to tab back and fourth through the menu or use the arrow keys. 

Escape exits. Tabbing after the last element of the menu should bring you to the bottom menu (emoji, image, embed). Tabbing back before first tab menu should bring you back to the editor.

In other words, we want to maintain the natural flow, except when we've got a menu available. We'll highjack the tabbing forward and then put it back where it was after the text.

**Comment by slafleche** [comment]
- [ ] Support keyboard navigation inside menu (both left/right as well as up/down)
- [ ] Escape exists menu 
- [ ] The paragraph level menu is conceptually closer to a radio button. We also need to update the appropriate aria attributes when one is selected.

**Comment by slafleche** [comment]
We're not going to use popper for now. We might later, but for now the editor works without it.
The UX problems are solved.
I've moved the accessibility issues into its own issue: https://github.com/vanilla/vanilla/issues/6806
